### PR TITLE
Add `as` property to VisuallyHidden

### PR DESCRIPTION
**Pull request by mareksuscak**
Thank you for contributing to Reach UI! Please fill in this template before submitting your PR to help us process your request more quickly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code (Compile and run).
- [x] Add or edit tests to reflect the change (Run with `yarn test`).
- [x] Add or edit Storybook examples to reflect the change (Run with `yarn start`).
- [x] Ensure formatting is consistent with the project's Prettier configuration.
- [x] Add documentation to support any new features.

This pull request:

- [ ] Creates a new package
- [ ] Fixes a bug in an existing package
- [x] Adds additional features/functionality to an existing package
- [ ] Updates documentation or example code
- [ ] Other

If creating a new package:

- [ ] Make sure the new package directory contains each of the following, and that their structure/formatting mirrors other related examples in the project:
  - [ ] `examples` directory
  - [ ] `src` directory with an `index.tsx` entry file
  - [ ] At least one example file per feature introduced by the new package
  - [ ] Base styles in a `style.css` file (if needed by the new package)

### CODE REVIEW

**Comment by mareksuscak** [comment]
For context, please see https://github.com/reach/reach-ui/issues/687

File: packages/visually-hidden/src/index.tsx [hunk 1]
```diff
@@ -11,41 +11,56 @@
...
-        whiteSpace: "nowrap",
-        wordWrap: "normal",
-      }}
-      {...props}
-    />
-  );
-});
+          // https://medium.com/@jessebeach/beware-smushed-off-screen-accessible-text-5952a4c2cbfe
+          whiteSpace: "nowrap",
+          wordWrap: "normal",
```
Comment (mareksuscak) [inline packages/visually-hidden/src/index.tsx:40]: Wonder if we need to add `display: inline;` in this list to maintain the expected behavior for block level elements.

Comment (chaance) [reply packages/visually-hidden/src/index.tsx:40]: I wouldn't worry about it. That said, it'd be worth spreading the `style` prop here if consumers have any issues they can solve with simple composition.

**Review by chaance — CHANGES_REQUESTED** [review]
Looks good to me. One quick change request and I'll get this merged. Appreciate you adding tests and docs!

File: packages/visually-hidden/src/index.tsx (hunk 1 above)
Comment (chaance) [reply packages/visually-hidden/src/index.tsx:40]: Went ahead and took care of this for you!

Comment (mareksuscak) [reply packages/visually-hidden/src/index.tsx:40]: I was going to take a look at it this morning but you beat me to it. Thank you @chaance!

**Pull request merged**